### PR TITLE
option: add openapiv2 support

### PR DESCRIPTION
**Pull request by cedric-parisi**
This PR adds OpenAPI v2 support:
- convert from proto to gunk by adding `openapiv2.Schema` and `openapiv2.Operation` to `+gunk` annotation
- generate from gunk to all.swagger.json 

minor fix on enum names by adding the enum type name to its value in case of duplicate

### CODE REVIEW

File: generate/generate.go [hunk 1]
```diff
@@ -762,6 +788,29 @@ func (g *Generator) methodOptions(method *ast.Field) (*desc.MethodOptions, error
 				return nil, err
 			}
 			g.addProtoDep("google/api/annotations.proto")
+		case "github.com/cedric-parisi/opt/openapiv2.Operation":
```
Comment (mvdan) [inline generate/generate.go]: surely this should be `gunk/opt` now?

File: loader/proto.go [hunk 2]
```diff
@@ -353,6 +422,13 @@ func (b *builder) handleEnum(e *proto.Enum) error {
 			continue
 		}
 
+		// Check if there is already an existing enum field with this name
+		if _, ok := b.existingDecls[ef.Name]; ok {
```
Comment (mvdan) [inline loader/proto.go]: simpler - `if b.existincDecls[ef.Name] {`

File: loader/proto.go [hunk 3]
```diff
@@ -429,10 +550,14 @@ func (b *builder) handleService(s *proto.Service) error {
 						}
 					default:
 						method = n
-						url, err = b.handleLiteralString(*l.Literal)
+						tmp, err := b.handleLiteralString(*l.Literal)
 						if err != nil {
 							return b.formatError(opt.Position, "option for %q should be a string (url)", method)
 						}
+						var re = regexp.MustCompile(`\{(.*?)\}`)
```
Comment (mvdan) [inline loader/proto.go]: compile this regex as a global, so you don't have to keep recomipiling it. also, prefer `x := value` over `var x = value` for local variables.

File: generate/generate.go (hunk 1 above)
Comment (cedric-parisi) [reply generate/generate.go]: indeed I missed it before submit...

File: .gitignore [hunk 4]
```diff
@@ -2,3 +2,5 @@
 /google
 /.bin
 /build/
+/pb/
+.gunkconfig
```
Comment (mvdan) [inline .gitignore]: don't add this, because we do have some gunk configs in examples and testdata

**Pull request merged**